corehq/apps/app_manager/management/commands/check_for_form_entities.py

```diff
@@ -0,0 +1,119 @@
+from datetime import datetime
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.core.paginator import Paginator
+from django.urls import reverse
+from corehq.apps.app_manager.models import Application, GlobalAppConfig
+from corehq.apps.app_manager.xform import parse_xml
+from corehq.apps.app_manager.exceptions import DangerousXmlException, XFormException
+from dimagi.utils.couch.database import iter_docs
+
+
+class Command(BaseCommand):
+    def __init__(self):
+        self.entities_found = 0
+        self.total_forms = 0
+
+    def add_arguments(self, parser):
+        parser.add_argument('--all', action='store_true')
+        parser.add_argument('--limit', default=0, type=int)
+
+    def handle(self, all, limit, *args, **kwargs):
+        if all:
+            print('Searching for entities in ALL forms')
+        else:
+            print('Searching for entities in CURRENT forms')
+
+        start_time = datetime.now()
+        self.process_forms(all, limit)
+        end_time = datetime.now()
+
+        elapsed = end_time - start_time
+
+        print(f'Found {self.entities_found} entities in {self.total_forms} forms')
+        print(f'Completed search in {elapsed}')
+
+    def process_forms(self, all, limit):
+        get_apps = get_all_apps if all else get_current_apps
+
+        for app in get_apps():
```

Comment:
Might be worth using [`with_progress_bar`](https://github.com/dimagi/commcare-hq/blob/90da907b7c70562ddece6772fac8c6051fa0d249/corehq/util/log.py#L260). That will require a separate query to determine the total number of objects in the iteration, but the user experience improvement is probably worth it if you end up running this over every app on production.

Reply:
Yeah, I was thinking this should probably be included, given how long this could run